Components/ANSIescapeCodeHandler.cpp

```diff
@@ -0,0 +1,248 @@
+**
+****************************************************************************/
+
+#include "ANSIescapeCodeHandler.h"
+
+namespace Utils {
+
+/*!
+    \class Utils::AnsiEscapeCodeHandler
+    \brief The AnsiEscapeCodeHandler class parses text and extracts ANSI escape codes from it.
+    In order to preserve color information across text segments, an instance of this class
+    must be stored for the lifetime of a stream.
+    Also, one instance of this class should not handle multiple streams (at least not
+    at the same time).
+    Its main function is parseText(), which accepts text and default QTextCharFormat.
+    This function is designed to parse text and split colored text to smaller strings,
+    with their appropriate formatting information set inside QTextCharFormat.
+    Usage:
+    \list
+    \li Create new instance of AnsiEscapeCodeHandler for a stream.
+    \li To add new text, call parseText() with the text and a default QTextCharFormat.
+        The result of this function is a list of strings with formats set in appropriate
+        QTextCharFormat.
+    \endlist
+*/
+
+AnsiEscapeCodeHandler::AnsiEscapeCodeHandler() :
+    m_previousFormatClosed(true)
+{
+}
+
+static QColor ansiColor(uint code)
+{
+    if (code < 8) return QColor();
+
+    const int red   = code & 1 ? 170 : 0;
+    const int green = code & 2 ? 170 : 0;
+    const int blue  = code & 4 ? 170 : 0;
+    return QColor(red, green, blue);
+}
+
+QList<FormattedText> AnsiEscapeCodeHandler::parseText(const FormattedText &input)
+{
+    enum AnsiEscapeCodes {
+        ResetFormat            =  0,
+        BoldText               =  1,
+        TextColorStart         = 30,
+        TextColorEnd           = 37,
+        RgbTextColor           = 38,
+        DefaultTextColor       = 39,
+        BackgroundColorStart   = 40,
+        BackgroundColorEnd     = 47,
+        RgbBackgroundColor     = 48,
+        DefaultBackgroundColor = 49
+    };
+
+    QList<FormattedText> outputData;
+
+    QTextCharFormat charFormat = m_previousFormatClosed ? input.format : m_previousFormat;
+
+    const QString escape = QLatin1String("\x1b[");
+    const int escapePos = input.text.indexOf(escape);
+    if (escapePos < 0) {
+        outputData << FormattedText(input.text, charFormat);
+        return outputData;
+    } else if (escapePos != 0) {
+        outputData << FormattedText(input.text.left(escapePos), charFormat);
+    }
+
+    const QChar semicolon       = QLatin1Char(';');
+    const QChar colorTerminator = QLatin1Char('m');
+    const QChar eraseToEol      = QLatin1Char('K');
+    // strippedText always starts with "\e["
+    QString strippedText = input.text.mid(escapePos);
+    while (!strippedText.isEmpty()) {
+        while (strippedText.startsWith(escape)) {
+            strippedText.remove(0, 2);
+
+            // \e[K is not supported. Just strip it.
+            if (strippedText.startsWith(eraseToEol)) {
+                strippedText.remove(0, 1);
```

Comment:
If you comment out just this single line you well see the letter k on both sides of text that you would expect to be colored. That means GCC is coloring its output using specific ANSI escapes which this ANSI escape handler does not support.